test/e2e/pivoting_test.go

```diff
@@ -24,146 +25,212 @@ import (
 
 	pivotingCluster := framework.DiscoveryAndWaitForCluster(ctx, framework.DiscoveryAndWaitForClusterInput{
-		Getter:    targetCluster.GetClient(),
-		Namespace: namespace,
-		Name:      clusterName,
-	}, e2eConfig.GetIntervals(specName, "wait-cluster")...)
+		Getter:    input.TargetCluster.GetClient(),
+		Namespace: input.Namespace,
+		Name:      input.ClusterName,
+	}, input.E2EConfig.GetIntervals(input.SpecName, "wait-cluster")...)
 
 	controlPlane := framework.GetKubeadmControlPlaneByCluster(ctx, framework.GetKubeadmControlPlaneByClusterInput{
-		Lister:      targetCluster.GetClient(),
+		Lister:      input.TargetCluster.GetClient(),
 		ClusterName: pivotingCluster.Name,
 		Namespace:   pivotingCluster.Namespace,
 	})
 	Expect(controlPlane).ToNot(BeNil())
 
 	By("Check that BMHs are in provisioned state")
 	waitForNumBmhInState(ctx, bmov1alpha1.StateProvisioned, waitForNumInput{
-		Client:    targetCluster.GetClient(),
-		Options:   []client.ListOption{client.InNamespace(namespace)},
+		Client:    input.TargetCluster.GetClient(),
+		Options:   []client.ListOption{client.InNamespace(input.Namespace)},
 		Replicas:  numberOfAllBmh,
-		Intervals: e2eConfig.GetIntervals(specName, "wait-object-provisioned"),
+		Intervals: input.E2EConfig.GetIntervals(input.SpecName, "wait-object-provisioned"),
 	})
 
 	By("Check if metal3machines become ready.")
 	waitForNumMetal3MachinesReady(ctx, waitForNumInput{
-		Client:    targetCluster.GetClient(),
-		Options:   []client.ListOption{client.InNamespace(namespace)},
+		Client:    input.TargetCluster.GetClient(),
+		Options:   []client.ListOption{client.InNamespace(input.Namespace)},
 		Replicas:  numberOfAllBmh,
-		Intervals: e2eConfig.GetIntervals(specName, "wait-object-provisioned"),
+		Intervals: input.E2EConfig.GetIntervals(input.SpecName, "wait-object-provisioned"),
 	})
 
 	By("Check that all machines become running.")
 	waitForNumMachinesInState(ctx, clusterv1.MachinePhaseRunning, waitForNumInput{
-		Client:    targetCluster.GetClient(),
-		Options:   []client.ListOption{client.InNamespace(namespace)},
+		Client:    input.TargetCluster.GetClient(),
+		Options:   []client.ListOption{client.InNamespace(input.Namespace)},
 		Replicas:  numberOfAllBmh,
-		Intervals: e2eConfig.GetIntervals(specName, "wait-machine-running"),
+		Intervals: input.E2EConfig.GetIntervals(input.SpecName, "wait-machine-running"),
 	})
 
 	By("PIVOTING TESTS PASSED!")
 }
 
-func installIronicBMO(targetCluster framework.ClusterProxy, isIronic, isBMO string) {
-	ironicTLSSetup := "IRONIC_TLS_SETUP"
-	ironicBasicAuth := "IRONIC_BASIC_AUTH"
+type installIronicBMOInput struct {
+	ManagementCluster          framework.ClusterProxy
+	BMPPath                    string
+	IsIronic                   string
+	IsBMO                      string
+	IronicTLSSetup             string
+	IronicBasicAuth            string
```

Comment:
It would make more sense to use bools instead of strings here IMO, and some of the names seem strange. Using bools also means you don't need the `isTrue` string to make the linter happy.
```suggestion
	BMOPath                    string
	InstallIronic              bool
	InstallBMO                 bool
	IronicTLSSetup             bool
	IronicBasicAuth            bool
```

Reply:
these [`IsBMO`, `IsIronic` ..] are set later as argument to `deploy.sh` not sure if used as `bool` will still keep the same value ? and since they were already typed as strings this PR is just moving vars to input struct not trying to introduce any other changes. for the new vars I like  `IsDeployment` I kept the same type style .. Can we do that in a follow up after merging these it would make more sense to me even BMO has depracated positional args for deploy.sh https://github.com/metal3-io/baremetal-operator/blob/cc171d0d8ee21107255db5a5239bb29f2afb0477/tools/deploy.sh#L58

Reply:
I'm not suggesting you should pass them to the script directly. But if you make a struct with input args, it makes more sense to use proper go types for it. Then when calling the script you can just [convert to strings](https://pkg.go.dev/strconv#FormatBool) inside the function instead of passing strings around in the go code.
To me it makes little sense adding the struct with all string values and then change it in a follow up. In that case just leave the struct out of this refactor and add it later. What do you think?

Reply:
I think since we are reading the sometime the value from environment vars : `IronicTLSSetup: input.E2EConfig.GetVariable(ironicTLSSetup)` we need to do more changes for now it is easier to keep them string we can add those changes as a follow up